src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/IssueFurtherEvidenceHandler.java

```diff
@@ -68,15 +68,35 @@ public void handle(CallbackType callbackType, Callback<SscsCaseData> callback) {
 
     private void issueFurtherEvidence(SscsCaseData caseData) {
         List<DocumentType> documentTypes = Arrays.asList(APPELLANT_EVIDENCE, REPRESENTATIVE_EVIDENCE, DWP_EVIDENCE, JOINT_PARTY_EVIDENCE, HMCTS_EVIDENCE);
-        List<FurtherEvidenceLetterType> allowedLetterTypes = Arrays.asList(APPELLANT_LETTER, REPRESENTATIVE_LETTER, JOINT_PARTY_LETTER);
-        documentTypes.forEach(documentType -> doIssuePerDocumentType(caseData, allowedLetterTypes, documentType));
+        List<FurtherEvidenceLetterType> allowedLetterTypes = Arrays.asList(APPELLANT_LETTER, REPRESENTATIVE_LETTER, JOINT_PARTY_LETTER, OTHER_PARTY_LETTER, OTHER_PARTY_REP_LETTER);
+        documentTypes.forEach(documentType -> doIssuePerDocumentType(caseData, allowedLetterTypes, documentType, null));
+
+        List<SscsDocument> groupedOtherPartyDocuments = groupUniqueOtherPartyDocumentsByOtherPartyId(caseData.getSscsDocument());
+
+        if (groupedOtherPartyDocuments != null && groupedOtherPartyDocuments.size() > 0) {
+            groupedOtherPartyDocuments.forEach(doc -> doIssuePerDocumentType(caseData, allowedLetterTypes, DocumentType.fromValue(doc.getValue().getDocumentType()), doc.getValue().getOriginalSenderOtherPartyId()));
+        }
+    }
+
+    private List<SscsDocument> groupUniqueOtherPartyDocumentsByOtherPartyId(List<SscsDocument> sscsDocuments) {
+
+        return sscsDocuments.stream()
+            .filter(doc -> OTHER_PARTY_EVIDENCE.getValue().equals(doc.getValue().getDocumentType()) || OTHER_PARTY_REPRESENTATIVE_EVIDENCE.getValue().equals(doc.getValue().getDocumentType()))
+            .filter(d -> "No".equals(d.getValue().getEvidenceIssued()))
+            .filter(distinctByKey(p -> p.getValue().getOriginalSenderOtherPartyId()))
+            .collect(Collectors.toList());
+    }
+
+    public static <T> Predicate<T> distinctByKey(Function<? super T, ?> keyExtractor) {
+        Set<Object> seen = ConcurrentHashMap.newKeySet();
+        return t -> seen.add(keyExtractor.apply(t));
     }
 
     private void doIssuePerDocumentType(SscsCaseData caseData, List<FurtherEvidenceLetterType> allowedLetterTypes,
```

Comment:
Not added by you now. but this method name is difficult to understand. The use of **do** and the number of parameters. It would be nice if we could wrap a few so we can keep it less than 2 params.

Reply:
Renamed the method - can't see an easy name of reducing parameter count so i'll leave it as this for now.